### PR TITLE
Fixes #33179 - Fix header in Enviroments page

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

**Comment by ofedoren** [comment]
@ezr-ondrej, shouldn't be that page extracted as part of Puppet removal?..

**Comment by ezr-ondrej** [comment]
> @ezr-ondrej, shouldn't be that page extracted as part of Puppet removal?..

indeed, that's why the fix is directly against 2.5 branch :)

**Review by ezr-ondrej — APPROVED** [review]
Looks good to me. //EDIT: JS error unrelated
@tbrisker ?

**Review by tbrisker — APPROVED** [review]
This doesn't seem like the correct fix and probably won't resolve the issue. The issue is that for generating the breadcrumb name, the mounter uses the BreadcrumbOption service to find the name:
https://github.com/theforeman/foreman/blob/f5a3dd2dedb569a0e6c5e8bb311c1da75906fc4f/app/helpers/layout_helper.rb#L76

Which in turn uses the menu manager to find the caption for the breadbrumb based on the controller name:

https://github.com/theforeman/foreman/blob/f5a3dd2dedb569a0e6c5e8bb311c1da75906fc4f/app/services/breadcrumbs_options.rb#L33

but when you have katello installed this is what is returned:
> Menu::Manager.get_resource_caption(:environments)
=> "Lifecycle Environments"

**Review by tbrisker — CHANGES_REQUESTED** [review]
oops, that should have been request changes, not approve :)

**Comment by tbrisker** [comment]
And the reason it finds that is that katello actually registers the lifecycle environment menu item with the same key:

https://github.com/Katello/katello/blob/7db3ca848ae1577810b40a879bc0675254226528/lib/katello/plugin.rb#L66

**Comment by tbrisker** [comment]
Alternatively, we could specify the breadcrumb explicitly like we do on the edit forms, by adding this to the view:
```
<%= breadcrumbs(:switchable => false,
                :items => [
                  { :caption => _('Puppet Environments'), :url => environments_path },
                ]) %>
```

**Comment by yifatmakias** [comment]
> Alternatively, we could specify the breadcrumb explicitly like we do on the edit forms, by adding this to the view:
> 
> ```
> <%= breadcrumbs(:switchable => false,
>                 :items => [
>                   { :caption => _('Puppet Environments'), :url => environments_path },
>                 ]) %>
> ```

To which view? the welcome page? I am not sure I fully understand

**Comment by tbrisker** [comment]
> > Alternatively, we could specify the breadcrumb explicitly like we do on the edit forms, by adding this to the view:
> > ```
> > <%= breadcrumbs(:switchable => false,
> >                 :items => [
> >                   { :caption => _('Puppet Environments'), :url => environments_path },
> >                 ]) %>
> > ```
> 
> To which view? the welcome page? I am not sure I fully understand

yes, if you add the `breadcrumbs` helper in the welcome.html.erb, it will override the default one which picks up the wrong name from the menu. Though I think that changing the menu item name in katello is a more correct solution to the issue.

Changing the titles from "Environments" to "Puppet Environments" still makes sense, but it won't resolve this issue and should be done in the puppet plugin rather than in the stable branch.

File: app/views/environments/welcome.html.erb [hunk 1]
```diff
@@ -1,2 +1,5 @@
-<% content_for(:title, _("Enviroments")) %>
```
Comment (tbrisker) [inline app/views/environments/welcome.html.erb:1]: This is the page title, it should still remain - but changed to Puppet Environments as well

**Review by tbrisker — APPROVED** [review]
Thanks @yifatmakias ! please open a PR to foreman_puppet adding this change to the plugin as well.

**Pull request merged**